src/domain/core/src/services/remote_aliases_registry.rs

```diff
@@ -38,31 +37,24 @@ pub enum GetAliasesError {
     ),
 }
 
-impl From<GetDatasetError> for GetAliasesError {
-    fn from(v: GetDatasetError) -> Self {
-        match v {
-            GetDatasetError::NotFound(e) => Self::DatasetNotFound(e),
-            GetDatasetError::Internal(e) => Self::Internal(e),
-        }
-    }
-}
-
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 // RemoteAliasResolver
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-#[async_trait]
+#[async_trait::async_trait]
 pub trait RemoteAliasResolver: Send + Sync {
     // Resolve remote push target.
     // Firstly try to resolve from AliasRegistry, if cannot do it
     // try to resolve via repository registry
     async fn resolve_push_target(
         &self,
-        local_dataset_handle: &DatasetHandle,
+        local_target: ResolvedDataset,
```

Comment:
Same here, likely better to continue to pass only the handle, not to surface internal concern of where the aliases are currently stored in the interface.

Reply:
Solved